hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileSystemContract.java

```
@@ -137,4 +142,30 @@ public void testRenameNonExistentPath() throws Exception {
         () -> super.testRenameNonExistentPath());
 
   }
+
+  @Test
+  public void testFileSystemCapabilities() throws Exception {
```

if the methods are on an interface only implemented by a few stores, there's no need to worry about s3a, abfs etc

---

yeah, that's why I chose to use `default` implementation when declaring the interface, then whoever does implement that will throw UnsupportedOperationException.

but still we should change those filesystem expectation, they're not extending or implementing those  interface for now. (we don't know if in the future anyone would like to build new feature on top, but it's very less likely for s3a and abfs)

---

but if we only have the interface implemented by those filesystems which actually do so (and viewfs + filterfs) then s3a, abfs etc don't care.

and viewfs and filterfs do need custom implementations; having default implementations hides this fact.